Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/details/QrCodeDetailViewModel.kt

```diff
@@ -29,74 +26,29 @@ class QrCodeDetailViewModel @AssistedInject constructor(
     private val traceLocationRepository: DefaultTraceLocationRepository
 ) : CWAViewModel() {
 
-    private val traceLocationFlow = MutableStateFlow<TraceLocation?>(null)
-    private val titleFlow = MutableStateFlow<String?>(null)
-    private val subtitleFlow = MutableStateFlow<String?>(null)
-    private val startTimeFlow = MutableStateFlow<Instant?>(null)
-    private val endTimeFlow = MutableStateFlow<Instant?>(null)
-    private val bitmapLiveData = MutableLiveData<Bitmap>()
+    private var traceLocation: TraceLocation? = null
+    private val mutableUiState = MutableLiveData<UiState>()
+    val uiState: LiveData<UiState>
+        get() = mutableUiState
+    val routeToScreen: SingleLiveEvent<QrCodeDetailNavigationEvents> = SingleLiveEvent()
 
     init {
-
         launch {
-            val traceLocation = traceLocationRepository.traceLocationForId(traceLocationId)
-
-            if (titleFlow.value == null) {
-                titleFlow.value = traceLocation.description
-            }
-
-            if (subtitleFlow.value == null) {
-                subtitleFlow.value = traceLocation.address
-            }
-
-            if (startTimeFlow.value == null) {
-                startTimeFlow.value = traceLocation.startDate
-            }
-
-            if (endTimeFlow.value == null) {
-                endTimeFlow.value = traceLocation.endDate
+            traceLocation = traceLocationRepository.traceLocationForId(traceLocationId).also {
+                mutableUiState.postValue(UiState(it))
```

Comment:
Please update the docs for this method since `qrCodeBitmap` is removed ?

Reply:
updated